detector/other/FieldMapBrBz.cpp

```diff
@@ -52,71 +72,54 @@ void FieldMapBrBz::fieldComponents(const double* pos , double* globalField) {
+  if(z < zMin  ) z = zMin;
+
   //APS: Note the mokka field map does not start at 0, so we have to assume that
   //this area is covered, or add some more parameters for the values where the
   //field is supposed to be. Now we just assume it starts at 0/0/0, outside there is no field
-  if (not ( 0.0 <= x && x <= rhoMax &&
-	    0.0 <= y && y <= zMax ) ) {
+  if (not ( rhoMin <= r && r <= rhoMax &&
+	    zMin   <= z && z <= zMax ) ) {
 
     return;
   }
 
-  if( x < rhoMin ) x = rhoMin;
-  if( y < zMin ) y = zMin ;
-
-
-  // Position of given point within region, normalized to the range
-  // [0,1]
-  const double xfraction = ( x - rhoMin ) / ( rhoMax - rhoMin );
-  const double yfraction = ( y - zMin )   / ( zMax   - zMin   );
+  int rBin,zBin;
+  double r0,z0;
+  double r1,z1;
 
-  // Need addresses of these to pass to modf below.
-  // modf uses its second argument as an OUTPUT argument.
-  double xdindex, ydindex;
+  rBin = int((r - rhoMin)/rhoStep);
+  zBin = int((z - zMin)/zStep);
+  r0   = rhoMin + rBin*rhoStep;
+  z0   = zMin   + zBin*zStep;
 
-  // Position of the point within the cuboid defined by the
-  // nearest surrounding tabulated points
-  const double xlocal = ( std::modf( xfraction * ( nRho - 1), &xdindex ) );
-  const double ylocal = ( std::modf( yfraction * ( nZ   - 1), &ydindex ) );
+  if(r0 > r) r0 -= rhoStep;
+  r1 = r0 + rhoStep;
+  if(z0 > z) z0 -= zStep;
+  z1 = z0 + zStep;
 
-  // The indices of the nearest tabulated point whose coordinates
-  // are all less than those of the given point
-  const int xindex = static_cast<int>(xdindex);
-  const int yindex = static_cast<int>(ydindex);
+  double rd = (r - r0)/(r1 - r0);
+  double zd = (z - z0)/(z1 - z0);
 
-  const int index0 =  xindex    * nZ + yindex   ;
-  const int index1 =  xindex    * nZ + yindex+1 ;
-  const int index2 = (xindex+1) * nZ + yindex   ;
-  const int index3 = (xindex+1) * nZ + yindex+1 ;
-
-  const FieldMapBrBz::FieldValues_t& fv0 = fieldMap[index0];
-  const FieldMapBrBz::FieldValues_t& fv1 = fieldMap[index1];
-  const FieldMapBrBz::FieldValues_t& fv2 = fieldMap[index2];
-  const FieldMapBrBz::FieldValues_t& fv3 = fieldMap[index3];
+  const FieldMapBrBz::FieldValues_t& B_r0z0 = fieldMap[GetBlobalIndex(r0,z0)];
+  const FieldMapBrBz::FieldValues_t& B_r1z0 = fieldMap[GetBlobalIndex(r1,z0)];
+  const FieldMapBrBz::FieldValues_t& B_r0z1 = fieldMap[GetBlobalIndex(r0,z1)];
+  const FieldMapBrBz::FieldValues_t& B_r1z1 = fieldMap[GetBlobalIndex(r1,z1)];
 
   double field[2] = {0.0, 0.0};
 
-  field[0] =
-    fv0.Br * (1-xlocal) * (1-ylocal)  +
-    fv1.Br * (1-xlocal) *    ylocal   +
-    fv2.Br *    xlocal  * (1-ylocal)  +
-    fv3.Br *    xlocal  *    ylocal   ;
-
-  field[1] =
-    fv0.Bz * (1-xlocal) * (1-ylocal)  +
-    fv1.Bz * (1-xlocal) *    ylocal   +
-    fv2.Bz *    xlocal  * (1-ylocal)  +
-    fv3.Bz *    xlocal  *    ylocal   ;
+  field[0] = (1.0 - rd)*(1.0 - zd)*B_r0z0.Br + rd*(1.0 - zd)*B_r1z0.Br + (1.0 - rd)*zd*B_r0z1.Br + rd*zd*B_r1z1.Br;
```

Comment:
Well, you make spaces around the operator inside the parenthesis, but then you don't leave any space between the parenthesis and the next operand, which makes things harder to read, because the "optical" grouping is just opposite of the operator grouping.